azkaban-common/src/main/java/azkaban/imagemgmt/version/VersionSetBuilder.java

```diff
@@ -50,7 +50,9 @@ public VersionSet build() throws IOException {
     ObjectMapper objectMapper = new ObjectMapper();
     String versionSetJsonString = objectMapper.writeValueAsString(versionSetElements);
     String versionSetMd5Hex = DigestUtils.md5Hex(versionSetJsonString);
-    int versionSetId = this.loader.getVersionSetId(versionSetMd5Hex, versionSetJsonString);
-    return new VersionSet(versionSetJsonString, versionSetMd5Hex, versionSetId);
+    final VersionSet versionSet;
+    versionSet = this.loader.getVersionSet(versionSetMd5Hex, versionSetJsonString)
+        .orElse(null);  // null implies Exception was thrown by the Dao Layer
+    return versionSet;
```

Comment:
Nit: return this.loader.getVersionSet(versionSetMd5Hex, versionSetJsonString)
        .orElse(null);